virttest/qemu_devices/qcontainer.py

```diff
@@ -2228,7 +2228,7 @@ def images_define_by_params(self, name, image_params, media=None,
     def serials_define_by_variables(self, serial_id, serial_type, chardev_id,
                                     bus_type=None, serial_name=None,
                                     bus=None, nr=None, reg=None,
-                                    bus_extra_params=''):
+                                    bus_extra_params='', reuse_bus=True):
```

Comment:
Instead of introducing a new argument/parameter, suggest reusing the existing one `bus` to represent the policy of creating a new bus, it can be done by introducing a special value, for example:
```
serial_bus = <new>
```

Reply:
Yes, it is one way, but i think the current change is smallest one.  To reuse the existing parameter `bus`, it is only a bus id defined by user, need change the code to create a qdevices and add to the `devices` list if the bus does not exist. Correct me if i am wrong, thanks @luckyh

Reply:
I mean something like:
```python
def serials_define_by_variables(...):
    ...
    if serial_type.startswith('virt'):
        if not bus or bus == "<new>":
            ...
            if bus != "<new>":
                bus = self.get_first_free_bus(...)
            ...
    ...
```
looks it would work.

Reply:
@luckyh   make sense, thanks!
@nanliu-r  Please check xuhan's comment above and update, thanks!

Reply:
Yes, it works when I define `serial_bus_vs* = 'new'`. which way would you perfer to? I'm ok with the two ways.
```
def serials_define_by_variables(...):
    ...
    if serial_type.startswith('virt'):
        if not bus or bus == "new":
            ...
            if bus != "new":
                bus = self.get_first_free_bus(...)
            ...
           if bus == "new" or bus is None or...
    ...
```

Reply:
@nanliu-r  drop the new param 'resue_bus', and use 'bus'

Reply:
ok, will update.